### PR TITLE
Blog:introduction to network role

### PR DESCRIPTION
**Pull request by liangwen12year**
Signed-off-by: Wen Liang <liangwen12year@gmail.com>

### CODE REVIEW

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+layout: post
+title: "Introduction to Network Role"
+section: Blog
+date: 2020-11-10T12:00:00
+author: Wen Liang
+category: talk
+---
+## Introduction
+
+The network role supports two providers: nm and initscripts. For CentOS/RHEL 6,
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:11]: "supports two providers: NetworkManager (nm) and initscripts."

The first time you introduce the abbreviation, spell it out so the reader knows what it is.  Then you can use `nm` in the rest of the document.

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+- `./defaults/ ` – Contains the default role configuration.
+- `./examples/` – Contains YAML examples for different configurations.
+- `./library/network_connections.py` – Contains the internal Ansible module,
+which is the main script. It controls the communication between the role and
+Ansible, imports the YAML configuration and applies the changes to the provider
+(i.e. NetworkManager, initscripts).
+- `./meta/` – Metadata of the project.
+- `./module_utils/network_lsr/` – Contains other files that are useful for the
+network role (e.g. the YAML argument validator)
+- `./tasks/` – Declaration of the different tasks that the role is going to execute.
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:40]: Each file matching `tests_*.yml` is a test playbook which is run by the CI system.

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+- `./tests/playbooks/` – Contains the complete tests for the role, `./tests/test_*.yml`
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:41]: tests/tests_*.yml

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+  * py26
+  * py27
+  * py36
+  * py37
+  * py38
+  * collection
+  * custom
+
+- tox, run all the tests
+- tox -e py36, test with Python 3.6, or just tox if you want to run all the tests.
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:65]: `py36` is pyunit testing with Python 3.6
omit ", or just tox if you want to run all the tests."

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+  * https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-2003.qcow2c
+  * https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2
+  * https://kojipkgs.fedoraproject.org/compose/cloud/
+- Install “standard-test-roles-inventory-qemu” package
+  ```bash
+  dnf install standard-test-roles-inventory-qemu
+  ```
+- [TEST_DEBUG=1] TEST_SUBJECTS=<image> ansible-playbook -v[v] -i <inventory file/script> <tests_….yml> 
+  ```bash
+  TEST_SUBJECTS=CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2 ansible-playbook -v -i /usr/share/ansible/inventory/standard-inventory-qcow2 tests/test_default.yml
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:83]: "tests_default.yml"

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+```
+
+#### Configuring Bonded Interface:
+
+```yaml
+network_connections:
+  - name: br0-bond0
+    type: bond
+    interface_name: bond0
+    master: internal-br0
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:141]: Has this been changed yet in the role to use more inclusive language?

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+      ca_cert: /etc/pki/tls/cacert.pem
+      domain_suffix_match: example.com
+```
+
+
+## Reference
+
+1. The external landing page for the system roles project, https://linux-system-roles.github.io/
+2. The external network role docs, https://github.com/linux-system-roles/network/
+3. The internal mojo information page, https://mojo.redhat.com/docs/DOC-1204325
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:298]: No redhat internal links, please

Comment (liangwen12year) [reply blog/_posts/2020-11-10-introduction-to-network-role.md]: already removed

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 8]
```diff
@@ -38,11 +38,11 @@ Ansible, imports the YAML configuration and applies the changes to the provider
 - `./module_utils/network_lsr/` – Contains other files that are useful for the
 network role (e.g. the YAML argument validator)
 - `./tasks/` – Declaration of the different tasks that the role is going to execute.
-- `./tests/playbooks/` – Contains the complete tests for the role, `./tests/test_*.yml`
-are shims to run tests once for every provider, `./tests/tasks/` contains task
-snippets that are used in multiple tests to avoid having the same code repeated
-multiple times.
-
+- `./tests/playbooks/` – Contains the complete tests for the role.
+- `./tests/test_*.yml` are shims to run tests once for every provider.
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:42]: `./tests/tests_*.yml`

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 9]
```diff
@@ -138,13 +138,13 @@ network_connections:
   - name: br0-bond0
     type: bond
     interface_name: bond0
-    master: internal-br0
+    controller: internal-br0
     slave_type: bridge
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:142]: what about "slave" - has that been changed yet?

File: blog/_posts/2020-11-10-introduction-to-network-role.md [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+- ...
+
+#### Integration Test
+
+- Download CentOS 6, CentOS 7, CentOS 8, Fedora images from
+  * https://cloud.centos.org/centos/6/images/CentOS-6-x86_64-GenericCloud-1907.qcow2c
+  * https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud-2003.qcow2c
+  * https://cloud.centos.org/centos/8/x86_64/images/CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2
+  * https://kojipkgs.fedoraproject.org/compose/cloud/
+- Install “standard-test-roles-inventory-qemu” package
```
Comment (richm) [inline blog/_posts/2020-11-10-introduction-to-network-role.md:77]: please use " 7-bit ascii character for double quote instead of “ and ”

Comment (liangwen12year) [reply blog/_posts/2020-11-10-introduction-to-network-role.md]: updated

File: blog/_posts/2020-11-10-introduction-to-network-role.md (hunk 3 above)
Comment (liangwen12year) [reply blog/_posts/2020-11-10-introduction-to-network-role.md]: updated

**Review by richm — APPROVED** [review]
lgtm

**Pull request merged**